### PR TITLE
New extension: ColorSlurp

### PR DESCRIPTION
**Pull request by Jordan-Ellis**
## Description

An extension that allows you to control [ColorSlurp's](https://colorslurp.com/) features from Raycast.

## Screencast

screenshot-1

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/colorslurp/package.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  "author": "Jordan-Ellis",
+  "categories": [
+    "Applications",
+    "Design Tools",
+    "Developer Tools"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "show-magnifier",
```
Comment (sxn) [inline extensions/colorslurp/package.json:16]: shouldn't this be `show-picker`? 🤔

**Comment by sxn** [comment]
hi @Jordan-Ellis - nice work w/ the extension!

One issue I've run into: when you don't have the application running, none of the commands work as intended; instead, running any command just results in the application being opened. 🤔

File: extensions/colorslurp/package.json (hunk 1 above)
Comment (Jordan-Ellis) [reply extensions/colorslurp/package.json:16]: The Magnifier in ColorSlurp is what allows you to pick colors from your screen, so “Pick Color” is supposed to be a shortcut to showing the magnifier.  I could rename it to “Show Magnifier“, to be more inline with what it does if you’d like. 

I realized that the name of the other command pick-color didn’t align with the title of “Edit Color“, so I just pushed a change that fixes the naming.

**Comment by Jordan-Ellis** [comment]
> One issue I've run into: when you don't have the application running, none of the commands work as intended; instead, running any command just results in the application being opened. 🤔

Hi @sxn! Looks like you found a bug with ColorSlurp, so I’ll have to release a ColorSlurp update. I already got it fixed in our dev version of ColorSlurp, so it’s not a problem with the extension. Thanks for pointing that out!

**Comment by sxn** [comment]
oh - I hadn't realised you're the creator of ColorSlurp! 😸 glad to hear the issue's been fixed. Let's merge the PR then - apart from that issue, the extension works great. 🎉

**Comment by Jordan-Ellis** [comment]
Awesome, thanks! 🎉

File: extensions/colorslurp/src/utilities.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    return bundleId === "com.IdeaPunch.ColorSlurp";
+  });
+}
+
+export async function openColorSlurpUrl(url: string) {
+  if (!(await isColorSlurpInstalled())) {
+    await showToast({
+      style: Toast.Style.Failure,
+      title: "ColorSlurp isn't installed",
+      message: "Install it from: https://colorslurp.com",
```
Comment (sxn) [inline extensions/colorslurp/src/utilities.ts:16]: one thing I realised just after merging - [Toasts](https://developers.raycast.com/api-reference/feedback/toast#toast) allow you to specify a `primaryAction` and `secondaryAction` that the user can invoke. So something like this

```suggestion
      message: "You can download it from the website or the app store",
      primaryAction: { title: "Open Homepage", onAction: () => open("https://colorslurp.com") },
      secondaryAction: {
        title: "Open App Store",
        onAction: () => open("https://apps.apple.com/us/app/colorslurp/id1287239339#?platform=mac"),
      },
```

would show up like this:

image

Not a big deal, but would make it ever so slightly simpler for users to access your website. 😸

Comment (Jordan-Ellis) [reply extensions/colorslurp/src/utilities.ts:16]: Oh that's super cool! Should I create a new pull request?

Comment (sxn) [reply extensions/colorslurp/src/utilities.ts:16]: go for it! 😉

**Pull request merged**